### ISSUE TITLE
Table or Column profile function / dmv

### ISSUE THREAD

**Issue by stobbea** [DaxStudio/DaxStudio#358]
It would be great to have something similar to what PowerQuery has from a profile standpoint. The things like rows, and distinct values I think I can already see in there, but more something like min, max, median, mode, nulls etc. 

I apologize if there's already something in there. I couldn't find it, and when googling, I found somebody else that asked for it specifically for calculated tables. Marco suggested that Vertipaq Analyzer could do something like that, but when I go into the integrated View Metrics, it doesn't give me anything like the above.

**Comment by dgosbell** [comment]
So we already have some of the basic information in the tooltip when you hover over a column
![image](https://user-images.githubusercontent.com/8241830/84845803-c8de1c00-b090-11ea-9065-7ff49e7406cd.png)

But we don't have anything that shows median, mode or nulls. And I think Vertipaq Analyzer only shows the distinct count of values.

But we could probably build something that would loop over every column in every table and gather this sort of information.

The simplest way to output this would probably be a table with columns something like the following:

| TableName | ColumnName | DataType | DistinctValues | Min | Max | Median | Mode | Nulls |

I'm not sure if it would make sense to add this to the Veripaq Analyzer functionality or if it should just be it's own button as this sort of information would probably be used differently to the output from Vertipaq Analyzer. I'll have to think a bit more about where a feature like this might fit.

**Comment by marcosqlbi** [comment]
VertiPaq Analyzer could implement this feature and it could display the info in a separate window called "Column profiling".
Assuming we prioritize the feature enough to be implemented, I'm not sure I'd enable this by default, because the memory and computational cost for getting the median and the mode would be very high. It should be an "on-demand" feature.

By "Nulls" do you mean counting the number of rows that have a BLANK value?

**Comment by stobbea** [comment]
As far as DAX is concerned, yes BLANK would stand in for null. For the work I publish in PowerBI myself I make heavy use of the PowerQuery profiling functionality, but when reviewing other team members' work, I often don't have the source data available, so can only look at the data once it's in the model, that's why this would be so helpful. 

Here's a screenshot of what PowerQuery gives me just on column selection - similar to what dgosbell pasted (I didn't realize even that was there - great start!)
![image](https://user-images.githubusercontent.com/39247173/84889217-8273c780-b066-11ea-91f8-eb67235b94eb.png)

And more like what I was thinking of, the Table.Profile in M returns this:
![image](https://user-images.githubusercontent.com/39247173/84889529-0037d300-b067-11ea-83f9-173fb8c0403e.png)

Thanks for your consideration!

**Comment by marcosqlbi** [comment]
Do you agree that mode and median are not required in table profiling? That is the most expensive part from a computational point of view - even though they could be useful in certain cases.

**Comment by stobbea** [comment]
Yes, I would consider them more 'nice to have'. The extremes are more important for finding data errors. Thanks

**Comment by dgosbell** [comment]
@stobbea how would you see this feature being used? 

I'm just wondering if you would review this data only on selected columns - so extending the tooltip would be one way of implementing this.

Or would you want to scan down a list of all the columns in the data model? In which case either extending the Vertipaq Analyzer view or building a dedicated button might make more sense.
